4-lesson

wanne27 opened the pull request:
нужно было достать html, и с помощью регулярок вытянуть из него ссылки, номера телефонов, email и записать в файл. Не знаю как сделать лучше...

wanne27 committed:
4 урок 2 задание

нужно было достать html, и с помощью регулярок вытянуть из него ссылки, номера телефонов, email, не знаю как сделать лучше...

wanne27 committed:
3 задание

заменить все предлоги на ГАВ, я пытался сделать одновременное считывание строки и перезапись ее, но чет не получилось, поэтому сначала считываю все строки в список стрингов, а потом меняю предлоги

wanne27 committed:
не в ту папку, исправил

wanne27 committed:
дополнительное задание

проверять логин и пароль, через регулярные выражения

wanne27 committed:
опять не в ту папку.... пофиксил

Izigraim commented on Ilya Khamichenok/004_Работа с текстом/Task2/Task2/Program.cs:
```diff
@@ -0,0 +1,39 @@
+
+            using (TextWriter tw = new StreamWriter("D:\\SavedList.txt"))
+            {
+                foreach (var matche in matchesURL)
```
Небольшой замечание по неймингу: единичный объект коллекции matchesURL будет URL, а не matche .

То же самое и относится к другим циклам.

Izigraim commented on Ilya Khamichenok/004_Работа с текстом/Task3/Task3/Program.cs:
```diff
@@ -0,0 +1,29 @@
+{
+    internal class Program
+    {
+        static void Main(string[] args)
```
Сделай так, чтобы у предлогов не учитывался их регистр.

Сейчас если предлог в файле написан с большой буквы, либо любая другая буква большая, то оно скипается и остается как было.

При этом учти ещё случай, что слова, которые не попадают под Regex, должны остаться такими, какими и были.

Izigraim commented on Ilya Khamichenok/004_Работа с текстом/TaskDop/TaskDop/Program.cs:
```diff
@@ -0,0 +1,46 @@
+{
+    internal class Program
+    {
+        static void Main(string[] args)
```
Давай в этой задачке сделаем код чуть более высокоуровневый.

Для этого создай отдельный static класс для валидации полей логин и пароль (например SignUpValidation).
В этом классе в private поля занеси паттерны для валидации, и создай 2 метода ValidateLogin и ValidatePassword, который в случае удачной валидации будут возвращаться true. И уже вот эти методы используй в условии цикла while, чтобы не писать while(true), а затем по какому-то условию делать break.

wanne27 committed:
пофиксил комментарии

wanne27 committed:
4 задание

есть файл чек, достаю данные из этого файла и вывожу на консоль в двух локалях, так же меняю цену в зависимости от курса доллара, курс доллара получаю с myfin, спарсил html с их конвертера валют

Izigraim commented on Ilya Khamichenok/004_Работа с текстом/Task4/Task4/ShowCheckList.cs:
```diff
@@ -0,0 +1,41 @@
+                    var price = Double.Parse(matchesPrice.Groups[1].Value.Replace(".", ","));
+                    var line = Regex.Replace(str, pricePattern, Math.Round(ExchangeCoure.BynToUsd(price), 2).ToString(enUsCulture));
+
+                    Console.WriteLine(Regex.Replace(line, belRubPattern, "$"));
```
Используй механизмы форматирования, которые изменяются на основе используемых региональных параметров, вместо строго вывода значка "доллара".

MSDN: https://learn.microsoft.com/ru-ru/dotnet/standard/base-types/formatting-types

wanne27 committed:
значок доллара добавляется из-за локали, а не вручную

бел.руб все равно пришлось менять на пустой string
